### PR TITLE
Add 'random' action and random_run config option

### PR DESCRIPTION
**Pull request by abjose**
and a small bugfix from a random run.

Ideally I would like the `random` action to not show up in the command list, but I'm not sure how to do that. I guess we could remove it from `adventurelib.commands` if not in a `random_run` game?

I was thinking we could use this as an integration test, but maybe we should just keep it in a branch.

### CODE REVIEW

**Comment by abjose** [comment]
Also this might make a fun AI

**Comment by flosincapite** [comment]
> Also this might make a fun AI

Gonna get the AlphaGo team to learn how to play text adventures next. That'll humble them :-p

File: cronenbroguelike/commands.py [hunk 1]
```diff
@@ -62,6 +64,31 @@ def look():
     _look()
 
 
+@when.when("random")
```
Comment (flosincapite) [inline cronenbroguelike/commands.py]: `adventurelib` registers any `@when`-ed function that has been defined. So if you don't want this command to show up without the `random_run` flag, you can just protect the function definition behind an `if` statement.

Note line 105 below, where `extra_commands` is imported. You could do something like that: just put this in a separate source file and selectively import that file.

Comment (abjose) [reply cronenbroguelike/commands.py]: Nice, thanks

**Comment by abjose** [comment]
Thanks for the review!
- moved `random` into extra_commands
- fixed a small `set changed size during iteration` bug

File: cronenbroguelike/commands.py [hunk 2]
```diff
@@ -293,6 +294,8 @@ def read(book):
     _, book = _find_available_item(book_name)
     if book is None:
         say.insayne(f"There is no {book_name} here to read.")
+    elif not isinstance(book, Book):
```
Comment (flosincapite) [inline cronenbroguelike/commands.py:297]: I like it!

**Review by flosincapite — APPROVED** [review]

**Pull request merged**